src/components/SearchField/SearchField.jsx

```diff
@@ -0,0 +1,137 @@
+import _ from 'lodash';
+import React, { createElement } from 'react';
+import { lucidClassNames } from '../../util/style-helpers';
+import { createClass, findTypes, omitProps } from '../../util/component-types';
+
+import TextField from '../TextField/TextField';
+import SearchIcon from '../Icon/SearchIcon/SearchIcon';
+
+import reducers from './SearchField.reducers';
+
+const cx = lucidClassNames.bind('&-SearchField');
+
+const {
+	bool,
+	func,
+	node,
+	number,
+	oneOfType,
+	string,
+} = React.PropTypes;
+
+/**
+* {"categories": ["controls", "text"], "madeFrom": ["TextField", "SearchIcon"]}
+*
+* This is a container that renders panels and controls its expansion/retraction.
+*/
+const SearchField = createClass({
+	displayName: 'SearchField',
+
+	components: {
+		TextField,
+		Icon: createClass({
+			displayName: 'SearchField.Icon',
+			propName: 'Icon',
+		}),
+	},
+
+	reducers,
+
+	propTypes: {
+		/**
+		 * Fires an event every time the user types text into the TextField.
+		 *
+		 * Signature: `(value, { event, props }) => {}`
+		 */
+		onChange: func,
+		/**
+		 * Fires an event when the user hits "enter" from the SearchField.
+		 *
+		 * Signature: `(value, { event, props }) => {}`
+		 */
+		onSubmit: func,
+		/**
+		 * Set the value of the input.
+		 */
+		value: oneOfType([
+			number,
+			string,
+		]),
+		/**
+		 * Disables the SearchField by greying it out.
+		 */
+		isDisabled: bool,
+		/**
+		 * placeholder value
+		 */
+		placeholder: string,
+		/**
+		* Appended to the component-specific class names set on the root
+		* element.
+		*/
+		className: string,
+		/**
+		* Passed through to the root element.
+		*/
+		Icon: node,
+	},
+
+	getDefaultProps() {
+		return {
+		};
+	},
+
+	render() {
+
+		const {
+			props,
+			props: {
+				className,
+				isDisabled,
+				onChange,
+				onSubmit,
+				placeholder,
+				value,
+				...passThroughs,
+			},
+		} = this;
+
+		const {
+			Icon,
+			TextField,
+		} = SearchField;
+
+		const textFieldProps = {
```

Comment:
these are the props i thought would be most useful for the default case. if you need more than this, you can just pass in your own `SearchField.TextField`, which is just a raw `TextField` with whatever props you want